### PR TITLE
feat: upgrade driver and bson to 5.0 MONGOSH-1358

### PR DESCRIPTION
**Pull request by addaleax**
(see also https://docs.google.com/document/d/1k0Ow5fd9bouBwboPzUisGN8lvP4YviNzDY4y2aI_TRQ/edit#)

##### chore: make --quiet and --verbose no-ops

These are no longer supported by the driver, and will only become
useful once full driver logging is implemented.

### CODE REVIEW

File: README.md
Comment (gribnoysup) [inline README.md:37]: Not sure about `verbose`, but `quiet` still makes difference in our code, for example the startup messages don't show up, right? If that's the case, we probably should keep these in the documentation 🤔

File: packages/java-shell/package.json
Comment (gribnoysup) [inline packages/java-shell/package.json:19]: Should this be webpack instead of browserify now? Not sure if this matters at all, as the base config we seem to require right from the root 🤔

File: packages/java-shell/package.json
Comment (gribnoysup) [inline packages/java-shell/package.json:14]: Is it intentional that we are running this in "development" and not "production" mode?

**Review by gribnoysup — APPROVED** [review]
Overall looks good! Left a few comments here and there

File: packages/java-shell/package.json
Comment (addaleax) [reply packages/java-shell/package.json]: No. :slightly_smiling_face: Fixed in 069806dcf1fb0c9d56d2d4c8c30dcf0ea79dd4ae

File: packages/java-shell/package.json
Comment (addaleax) [reply packages/java-shell/package.json]: This was just a lazy way of getting all the browserify polyfills in place :slightly_smiling_face: Switched to an explicit list in 069806dcf1fb0c9d56d2d4c8c30dcf0ea79dd4ae

**Comment by addaleax** [comment]
Just for context, we should wait for https://jira.mongodb.org/browse/NODE-5052 and probably other tickets from https://jira.mongodb.org/issues/?jql=labels%20%3D%20node-v5-support to be resolved before merging this.

**Comment by addaleax** [comment]
@kornilova203 Just fyi, we’re looking to publish a new release with the changes here, so we’ll probably merge this tomorrow unless that seems like a bad idea to you – the java-shell tests do seem to pass just fine in CI

**Comment by kornilova203** [comment]
@addaleax, now tests pass on my machine too :) don't know what was the problem

**Pull request merged**
